File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        max_len = 0

        arr = []

        for right in range(len(s)):
            
            arr.append(s[right])

            while len(arr) > len(set(arr)):
                arr.remove(s[left])
                left += 1

            max_len = max(max_len, len(arr))

        return max_len

File: Data Structures & Algorithms/depth-of-binary-tree/submission-5.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        let depth = 0

        function dfs(root, depth){
            if (!root){
                return depth
            }
            let left = dfs(root.left, depth + 1)
            let right = dfs(root.right, depth + 1)
            return Math.max(left, right)
        }

        return dfs(root, depth)
    }
}
